File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n=nums.length;
        int []left =new int[n];
        int []right=new int[n];
        left[0]=nums[0];
        right[n-1]=nums[n-1];
        for(int i=1;i<n;i++)
        {
            left[i]=left[i-1]*nums[i];
            right[n-i-1]=right[n-i]*nums[n-i-1];
            // int x=n-i-1;
            // System.out.println("left : "+ i+" " + left[i] + " Right : "+ x+" " +right[n-i-1]);
        }
        
        int [] res=new int[n];
        for(int i=0;i<n;i++)
        {
            if(i==0) res[i]=right[i+1];
            else if (i==n-1) res[i]=left[i-1];
            else
                res[i]=right[i+1]*left[i-1];
        }
        return res;
    }
}  

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:

        if not root:
            return []
        res = [] 
        queue = collections.deque()
        queue.append(root)


        while queue:
            qlen = len(queue)
            level = []
            for i in range(qlen):
                node= queue.popleft()
                if node:
                    level.append(node.val)
                    queue.append(node.left)
                    queue.append(node.right)
            if level:
                res.append(level[-1])

        return res